Widgets/Widgets/OnThisDayWidget.swift

```diff
@@ -38,7 +38,14 @@ struct OnThisDayProvider: TimelineProvider {
     func getTimeline(in context: Context, completion: @escaping (Timeline<OnThisDayEntry>) -> Void) {
         dataStore.fetchLatestAvailableOnThisDayEntry { entry in
             let currentDate = Date()
-            let timeline = Timeline(entries: [entry], policy: .after(currentDate.dateAtMidnight() ?? currentDate))
+            let nextUpdate: Date
+            if entry.error == nil {
```

Comment:
I think we can just skip the error checking and move to a 2 or 3 hour update interval for all the widgets and see how far that gets us. I can see the potential value of doing the error checking first, but I wonder if iOS decides to perform timeline updates based on our policy _and_ its own mix of policy decisions like network reachability.

I think our eventual ideal model of constructing the timeline may be something like:
- First check network reachability. If unavailable, and we have a sufficient budget of widget updates, schedule another update for after a few minutes. If we don't have a sufficient budget of widget updates, just deliver cached/placeholder data with another update scheduled for after a couple hours.
- If network is reachable, perform a fetch. 
- If fetched content matches our criteria (for the current day, not placeholder entry, no errors), present it and schedule a timeline update at midnight. Otherwise, present what's available, and schedule another update after a few minutes as the widget update budget allows. 
- Have a local cache recording repeated "after a few minutes" attempts to be sure we're not burning through our available widget update budget

But I think maybe that's a bit too much before we get a better understanding of widgets in use to see where the problems are. So for now maybe just: 
```swift
let components = DateComponents(hour: 2)
let nextUpdate = Calendar.current.date(byAdding: components, to: currentDate) ?? currentDate
let timeline = Timeline(entries: [entry], policy: .after(nextUpdate))
```
Just moving everything to this 2 or 3 hour update interval will help us cover our stale data cases as well.

What do you think?

Reply:
I really, really like your eventual model plan.

I'm open to the idea for ignoring error checking, but I'm curious what the value is in refreshing if we're fairly confident there was no error. It seems like we don't actually know our budget of widget updates, and there is no way to query for it. (Though I could have missed that somewhere in documentation, it seemed like it was a bit of a black box, like on WatchOS.) While there is potentially no downside to spamming a refresh every 2-3 hours (if we run out of our budget, we presumably already have our latest version), I'm not sure we're guaranteed that budget resets at midnight. If they use something other than a hard midnight cut (which we can't know), we may not get a refresh at midnight if we spammed too much during the day.

Reply:
The main things I think a blanket 2-3 hour refresh would afford us:

1. simpler conceptual model (no branching based on error), so as we move forward into understanding more about these widgets in production use we know they are always (as much as we can hope) getting updates on a 2-3 hour interval
2. covering more stale or insufficient data cases - for example, the absence of image data on a first pass of top read would be ok and not considered an error, but if network conditions improve/the system is allowed to retry with more frequency, the fidelity of the presented data could be improved to include the images.

If you feel strongly about your current approach, let's go for it, but if you're on the fence about it, I think the blanket 2-3 hour approach could be a good alternative to start. Up to you!

Reply:
I think I went from neutral to favoring the "only update if problems" approach while writing my last comment. (Apologies - I think my writing starts with neutrality, but doesn't quite end there.)

The most important thing in my opinion is to ensure we don't burn through our budget and not get a requested midnight update (after the day turns). My concern about every 2-3 hours is that we don't know our budget, and can't be confident when the budget resets. So if for some reason we burn through and don't get a midnight refresh, the widget will look out of date to the users.

If there is documentation around the budget that renders my concerns moot, I'm happy to go with the every 2-3 hours approach. But if we can't be sure, I think I'd rather err on the safe side.

Reply:
Ok sounds good. While I'm pretty certain I've seen/heard some official source suggest something to the effect of having a more liberal budget than what we have here, I'm not having good luck finding that source again. So safe like you suggest sounds good. Thanks for exploring my alternative a bit.